### PR TITLE
ui: pki import key

### PR DESCRIPTION
**Pull request by hellobontempo**
import

### CODE REVIEW

File: ui/lib/pki/addon/components/pki-key-form.js [hunk 1]
```diff
@@ -46,6 +46,11 @@ export default class PkiKeyForm extends Component {
     }
   }
 
+  @action
+  onFileUpload() {
+    // do something
```
Comment (kiannaquach) [inline ui/lib/pki/addon/components/pki-key-form.js:51]: Just wanted to make sure this action isn't supposed have anything yet?

File: ui/lib/pki/addon/components/pki-key-import.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  @tracked modelValidations;
+
+  @task
+  *submitForm(event) {
+    event.preventDefault();
+    try {
+      const { keyName } = this.args.model;
+      yield this.args.model.save({ adapterOptions: { import: true } });
+      this.flashMessages.success(`Successfully imported key ${keyName}.`);
+      this.args.onSave();
```
Comment (kiannaquach) [inline ui/lib/pki/addon/components/pki-key-import.js:39]: this is an awesome way to do this!!

File: ui/lib/pki/addon/components/pki-key-import.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+export default class PkiKeyImport extends Component {
+  @service store;
+  @service flashMessages;
+
+  @tracked errorBanner;
+  @tracked invalidFormAlert;
+  @tracked modelValidations;
+
+  @task
+  *submitForm(event) {
```
Comment (kiannaquach) [inline ui/lib/pki/addon/components/pki-key-import.js:33]: Not sure if this necessary now, but I've seen some concurrency tasks with `waitFor` and I think it's for tests in the future!

**Review by kiannaquach — COMMENTED** [review]
Nothing blocking! Just had a few comments and wanted to make sure the action in `ui/lib/pki/addon/components/pki-key-form.js` is supposed to be empty. :) I also noticed this PR is still a draft

File: ui/app/adapters/pki/key.js [hunk 4]
```diff
@@ -4,8 +4,13 @@ export default class PkiKeyAdapter extends ApplicationAdapter {
...
 
   createRecord(store, type, snapshot) {
-    const { record } = snapshot;
-    const url = this.getUrl(record.backend) + '/generate/' + record.type;
+    const { record, adapterOptions } = snapshot;
+    let url = this.getUrl(record.backend);
+    if (adapterOptions.import) {
+      url = url + '/import';
+    } else {
+      url = url + '/generate/' + record.type;
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/adapters/pki/key.js:12]: yeah! Great idea - I just saw Chelsea did this in her pr and thought it read so much clearer 🤩

File: ui/lib/pki/addon/components/pki-key-form.js (hunk 1 above)
Comment (hellobontempo) [reply ui/lib/pki/addon/components/pki-key-form.js:51]: 🤔  nope, I definitely added something - maybe I'm missing a commit

File: ui/lib/pki/addon/components/pki-key-import.js (hunk 3 above)
Comment (hellobontempo) [reply ui/lib/pki/addon/components/pki-key-import.js:33]: I'd wondered the purpose in adding that! Thank you!

File: ui/lib/pki/addon/components/pki-key-form.js (hunk 1 above)
Comment (hellobontempo) [reply ui/lib/pki/addon/components/pki-key-form.js:51]: Hmm - it's showing up on my end! Maybe you were so speedy it was an outdated github version?!

Comment (hellobontempo) [reply ui/lib/pki/addon/components/pki-key-form.js:51]: OOH 🤣 this should be deleted, all of the import functionality is handled by the `pki-key-import.js` component. Holiday brain 🌴 

nice catch!

**Review by kiannaquach — APPROVED** [review]

**Pull request merged**